btk/measure.py

```diff
@@ -135,18 +137,107 @@ def basic_measure(
     return {"catalog": catalog}
 
 
-def sep_measure(
+def sep_multiband_measure(
     batch,
     idx,
     channels_last=False,
     surveys=None,
+    matching_threshold=1.0,
     sigma_noise=1.5,
     is_multiresolution=False,
     **kwargs,
 ):
-    """Return detection, segmentation and deblending information with SEP.
+    """Returns centers detected with source extractor by combining predictions in different bands.
+
+    NOTE: If this function is used with the multiresolution feature,
+    measurements will be carried on the first survey.
+
+    Args:
+        batch (dict): Output of DrawBlendsGenerator object's `__next__` method.
+        idx (int): Index number of blend scene in the batch to preform
+            measurement on.
+        sigma_noise (float): Sigma threshold for detection against noise.
+        matching_threshold (float): Match centers of objects that are closer than
+            this threshold to a single prediction.
+
+    Returns:
+            dict containing catalog with entries corresponding to measured peaks.
+    """
+    channel_indx = 0 if not channels_last else -1
+    # multiresolution
+    if is_multiresolution:
+        if surveys is None:
+            raise ValueError("surveys are required in order to use the MR feature.")
+        survey_name = surveys[0].name
+        image = batch["blend_images"][survey_name][idx]
+        wcs = batch["wcs"][survey_name]
+
+    # single-survey
+    else:
+        image = batch["blend_images"][idx]
+        wcs = batch["wcs"]
+
+    # run source extractor on the first band
+    band_image = image[0] if channel_indx == 0 else image[:, :, 0]
+    bkg = sep.Background(band_image)
+    catalog = sep.extract(band_image, sigma_noise, err=bkg.globalrms, segmentation_map=False)
+
+    # convert predictions to arcseconds
+    ra_coordinates, dec_coordinates = wcs.pixel_to_world_values(catalog["x"], catalog["y"])
+    ra_coordinates *= 3600
+    dec_coordinates *= 3600
+
+    # iterate over remaining bands and match predictions using KdTree
+    for band in range(1, image.shape[channel_indx]):
+        # run source extractor
+        band_image = image[band] if channel_indx == 0 else image[:, :, band]
+        bkg = sep.Background(band_image)
+        catalog = sep.extract(band_image, sigma_noise, err=bkg.globalrms, segmentation_map=False)
+
+        # convert predictions to arcseconds
+        ra_detections, dec_detections = wcs.pixel_to_world_values(catalog["x"], catalog["y"])
+        ra_detections *= 3600
+        dec_detections *= 3600
+
+        # convert to sky coordinates
+        c1 = SkyCoord(ra=ra_detections * units.arcsec, dec=dec_detections * units.arcsec)
+        c2 = SkyCoord(ra=ra_coordinates * units.arcsec, dec=dec_coordinates * units.arcsec)
+
+        # add new coordinates
+        if len(c1) > 0 and len(c2) > 0:
+            idx, d2d, d3d = c1.match_to_catalog_sky(c2)
```

Comment:
what exactly are the `d2d` and `d3d` ?

Reply:
In general `match_to_ctalog_sky` matches each pair of coordinates in `c1` to k-nearest neighbours in `c2`. In our case, we are only interested in the closest coordinate, so `k=1` 

- `d2d` - is a matrix with the shape `[len(c1), k]` that contains angular distances on the sky between each coordinate in `c1` and its neighbors.

- `d3d` - is the same as `d2d` but it computed 3d distance between objects if `z` was provided, otherwise it assumes that all coordinates are equally far away from us.

I added few comments and changed `d2d` to `distance2d` to make code more readable.

Reply:
hmm I don't think I see these changes? perhaps you forgot to push them?

Reply:
ah it was me looking at the wrong diff 😆 , this looks great now